activesupport/CHANGELOG.md

```diff
@@ -1,3 +1,19 @@
+*   Fix a regression with division by a Duration object
+
+    Before:
+        rental_end = DateTime.parse('01-01-2017 12:45:00').utc.in_time_zone('Europe/Berlin')
+        rental_start = DateTime.parse('01-01-2017 12:00:00').utc.in_time_zone('Europe/Berlin')
+        (rental_end - rental_start) / 1.hour
+        => 2700.0 hours
+    
+    After:
+        rental_end = DateTime.parse('01-01-2017 12:45:00').utc.in_time_zone('Europe/Berlin')
+        rental_start = DateTime.parse('01-01-2017 12:00:00').utc.in_time_zone('Europe/Berlin')
+        (rental_end - rental_start) / 1.hour
+        => 0.75
```

Comment:
@bxt the issue here is that a DateTime minus a DateTime gives you a duration in days compared to the seconds you get when you subtract two Time instances. ActiveSupport::Duration assumes that dimensionless numbers are seconds so it's not doing what you think it is. If you wrap the subtraction with a `().days` and then do the division and call `to_i` on the result you will get the right value. However I do agree that a duration divided by a duration should be dimensionless number